fabric-textures-v0/src/main/java/net/fabricmc/fabric/api/client/texture/ColorSniffer.java

```diff
@@ -1,32 +1,99 @@
-package net.fabricmc.fabric.impl.client.texture;
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.texture;
 
 import net.fabricmc.fabric.mixin.client.texture.SpriteAccessor;
 import net.minecraft.client.texture.NativeImage;
 import net.minecraft.client.texture.Sprite;
 
+import java.awt.*;
+
 /**
- * Utility class for averaging Sprite colors.
- * You should call one of these methods!
+ * Utility class for averaging Sprite colors, or using the provided Accessor
+ * through {@linkplain ColorSniffer#getImagesForSprite(Sprite)}.
  */
-public class ColorSniffer {
+public final class ColorSniffer {
+	private static final int errorColor = 0xFFFF00FF;
+
+	private ColorSniffer() { }
 
 	/**
+	 * Routing method for calculating the average color of a Sprite.
+	 *
+	 * <p>Assumes the provided Sprite only has one frame.
+	 * Otherwise, {@see ColorSniffer#getAverageSpriteColorForFrame(Sprite, int, AveragingMethod)}
+	 *
 	 * @param sprite The Sprite you want to get the average color from.
-	 * @return The average ABGR color of the provided Sprite.
+	 * @param averagingMethod The method used to calculate the returned color.
+	 * @return The average ARGB color of the provided Sprite. Returns {@linkplain ColorSniffer#errorColor} if something goes wrong.
 	 */
-	public static int getAverageSpriteColor(Sprite sprite) { return getAverageSpriteColor(sprite, 0); }
+	public static int getAverageSpriteColor(Sprite sprite, AveragingMethod averagingMethod) {
```

Comment:
Just expose the `calcuateAverageRgb/HsvSpriteColor` methods, no need to switch on an enum

Reply:
I feel like people will just grab the rgb method thinking `getAverageHsv` returns something they can't pass on

Reply:
🤤 just specify that in javadoc that both return rgb color but the decomposition is different

Reply:
you're implying that people can read before using